### PR TITLE
Generating recommendataions 

### PR DESCRIPTION
**Pull request by msvinaykumar**
WIP code

### CODE REVIEW

File: src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+                List<String> promQls = new ArrayList<>();
+                promQls.add("sum(rate(container_cpu_usage_seconds_total{container!~\"POD|\",namespace=\"%s\",container=\"%s\"}[%sm])) by (namespace,container)");
+
+                promQls.forEach(ql -> {
+                    String promQL = String.format(ql,"ros-prod","kruize-recommendations","15");
+                    System.out.println(promQL);
+                    String podMetricsUrl = null;
+                    try {
+                        podMetricsUrl = String.format("%s?query=%s&start=%s&end=%s&step=%s",
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java:154]: all this values will derive from KruizeObject

File: src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+                    kruizeObject = mainKruizeExperimentMAP.get(experiment_name);
+                }
+                // Get DataSource and connect
+                // Get MetricsProfile name and list of promQL to fetch
+                // Find LongTerm duration  keep Start , End and Step(measurement Duration)
+                // prepare <List>ExperimentResultData
+                // and call new ExperimentInitiator().generateAndAddRecommendations(kruizeObject, experimentResultDataList, interval_start_time, interval_end_time);    // TODO: experimentResultDataList not required
+
+                List<String> promQls = new ArrayList<>();
+                promQls.add("sum(rate(container_cpu_usage_seconds_total{container!~\"POD|\",namespace=\"%s\",container=\"%s\"}[%sm])) by (namespace,container)");
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java:147]: This Promqls will be constant and will be derived from Metrics profile , 16 more promql will be added

File: src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+import static com.autotune.analyzer.utils.AnalyzerConstants.ServiceConstants.CHARACTER_ENCODING;
+import static com.autotune.analyzer.utils.AnalyzerConstants.ServiceConstants.JSON_CONTENT_TYPE;
+import static com.autotune.analyzer.utils.AnalyzerErrorConstants.AutotuneObjectErrors.MISSING_EXPERIMENT_NAME;
+import static com.autotune.analyzer.utils.AnalyzerErrorConstants.AutotuneObjectErrors.MISSING_INTERVAL_END_TIME;
+
+/**
+ *
+ */
+@WebServlet(asyncSupported = true)
+public class GenerateRecommendations extends HttpServlet {
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java:71]: Eventually This class will be compatible with ROS as well as kruizeLocal.

File: src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+            try {
+                long interval_end_time_epoc = interval_end_time.getTime() / 1000 - (interval_end_time.getTimezoneOffset() * 60);
+                ;
+                long interval_start_time_epoc = interval_start_time.getTime() / 1000 - (interval_start_time.getTimezoneOffset() * 60);
+                ;
+
+                new ExperimentDBService().loadExperimentFromDBByName(mainKruizeExperimentMAP, experiment_name);
+                if (null != mainKruizeExperimentMAP.get(experiment_name)) {
+                    kruizeObject = mainKruizeExperimentMAP.get(experiment_name);
+                    dataSourceInfo = new ExperimentDBService().loadDataSourceFromDBByName(kruizeObject.getDataSource());
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java:145]: @khansaad 
Could you please confirm whether the experiment data source is being retrieved from the database? It appears that I was receiving null values.

Comment (khansaad) [reply src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java:145]: Updated the converter method to include the datasource value. Let me know if it's working now

**Review by dinogun — APPROVED** [review]
Hackathon Merge!

**Pull request merged**